collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -800,13 +804,15 @@ protected void onActivityResult(int requestCode, int resultCode, final Intent in
             case RequestCodes.EX_STRING_CAPTURE:
             case RequestCodes.EX_INT_CAPTURE:
             case RequestCodes.EX_DECIMAL_CAPTURE:
-                String key = "value";
-                boolean exists = intent.getExtras().containsKey(key);
-                if (exists) {
-                    Object externalValue = intent.getExtras().get(key);
-                    if (getCurrentViewIfODKView() != null) {
-                        setWidgetData(externalValue);
-                    }
+                Object externalValue = externalAppIntentProvider.getValueFromIntent(intent);
+                if (getCurrentViewIfODKView() != null) {
+                    setWidgetData(externalValue);
+                }
+                break;
+            case RequestCodes.EX_ARBITRARY_FILE_CHOOSER:
+                externalValue = externalAppIntentProvider.getValueFromIntent(intent);
```

Comment:
We'd said that since a `ClipData` needs to be provided anyway, we'll just take whatever is in the `ClipData` and use that, forgetting about any extras. From [the spec doc](https://docs.google.com/document/d/1JbKiwgBCuI2t3Cd8n5OnyrNce3woq20ncZbN3oGcPhU/edit): "Collect will read the ClipData contents and do the same as above, except that it will ignore any extras provided as well as the ClipData filename. Additionally, if there’s more than one item in the ClipData, Collect will fail silently and clear the field."

I'm not too picky about what to do in the case of multiple `ClipData` items as long as it's explicitly documented. If clearing the field is not practical, we could also just always take the first one. If no `ClipData` is provided, the ideal would be to clear the field if there's already an answer. Silently failing would be ok (though seems straightforward to clear the field?).

Reply:
Ok I changed the code and now it is using ClipData. I also added removing an existing answer if no ClipData is provided as you said.